test-data/unit/check-based-literal-types.test

```diff
@@ -0,0 +1,79 @@
+[case testBareLiteralLiterals]
+from __future__ import annotations
+
+a: 1
+a2: 1 | str
+reveal_type(a)  # N: Revealed type is "1"
+
+b: True
+b2: True | str
+reveal_type(b)  # N: Revealed type is "True"
+
+c: False
+c2: False | str
+reveal_type(c)  # N: Revealed type is "False"
+
+d: 1.1  # E: Invalid type: float literals cannot be used as a type  [valid-type]
+e2: 1.1 | str  # E: Invalid type alias: expression is not a valid type  [valid-type]
+
+e: 1j  # E: Invalid type: complex literals cannot be used as a type  [valid-type]
+e2: 1j | str  # E: Invalid type: complex literals cannot be used as a type  [valid-type]
+
+[builtins fixtures/tuple.pyi]
+
+
+[case testNoBareLiteralLiterals]
+a1: 1
+a2: 1 | str  # E: "1" is a bare literal and shouldn't be used in a type operation without "__future__.annotations"  [valid-type]
+
+b1: True
+b2: True | str  # E: "True" is a bare literal and shouldn't be used in a type operation without "__future__.annotations"  [valid-type]
+
+c1: False
+c2: False | str  # E: "False" is a bare literal and shouldn't be used in a type operation without "__future__.annotations"  [valid-type]
+
+d1: 1.1  # E: Invalid type: float literals cannot be used as a type  [valid-type]
+d2: 1.1 | str  # E: Invalid type alias: expression is not a valid type  [valid-type]
+
+e1: 1j  # E: Invalid type: complex literals cannot be used as a type  [valid-type]
+e2: 1.1 | str  # E: Invalid type alias: expression is not a valid type  [valid-type]
+
+
+[case testBareLiteralEnum]
+from __future__ import annotations
+from enum import Enum
+
+class E(Enum):
+    A = 1
+
+a: E.A
+a2: E.A | str
+reveal_type(a)  # N: Revealed type is "__main__.E.A"
+[builtins fixtures/tuple.pyi]
+
+[case testNoBareLiteralEnum]
+from enum import Enum
+
+class E(Enum):
+    A = 1
+
+a: E.A
+a: E.A | str  # E: Invalid type: try using Literal[E.A] instead?  [misc]
```

Comment:
shouldn't this error be `valid-type`

Reply:
uhhh, yeah lol

Reply:
but that's a breaking change, I don't want to touch it here, maybe up stream.

Reply:
isn't that the point of running tests in based mode and non-based mode?

Reply:
nah, it's like an existing error that is broken upstream